### PR TITLE
Enrich the logs streamed in tests with useful information.

### PR DESCRIPTION
**Pull request by markusthoemmes**
- Level is very useful to spot errors at a glance.
- Time is very helpful to be able to order the statements and notice time gaps.
- Key helps identifying which ressource is being reconciled (could be a child resource).

/assign @mattmoor 

Let's discuss here if this is what we want/need for now. Just printing time without ordering things is not ideal, but maybe a good first step?

### CODE REVIEW

File: test/logstream/kubelogs.go [hunk 1]
```diff
@@ -47,6 +47,9 @@ type logger func(string, ...interface{})
 
 var _ streamer = (*kubelogs)(nil)
 
+// timeFormat defines a simple timestamp with millisecond granularity
+const timeFormat = "2006-01-02 15:04:05.000"
```
Comment (vagababov) [inline test/logstream/kubelogs.go]: Do we want the date, in reality? The test run times are usually bound to the day, but this increases the line length and adds noise.
I like the way glog does formatting
[FirstCharOfLevel][DDMM][HHMMSS.SSS] -> [0201] 150405.000.
Here of course we can just keep `[FirstCharOfLevel]150405.000`.

Comment (markusthoemmes) [reply test/logstream/kubelogs.go]: Points taken, I adjusted the time format and reduced the log level to only the first char. I kept `:` in the time. If people feel strongly, I can remove those as well, but I feel like they add to readability a lot.

**Pull request merged**